test/controllers/manageBusinesses/cease/IncomeSourceEndDateControllerSpec.scala

```diff
@@ -556,22 +556,32 @@ class IncomeSourceEndDateControllerSpec extends TestSupport with MockAuthenticat
         s"SelfEmployment - unable to set session data ${CeaseIncomeSourceData.dateCeasedField}" when {
           "called .submit" in {
             setupMockCreateSession(true)
-            setupMockSetSessionKeyMongo(CeaseIncomeSourceData.dateCeasedField)(Left(new Exception()))
+            val testMap: Map[String, String] = Map(
+              CeaseIncomeSourceData.dateCeasedField -> "2022-08-27"
+            )
+            setupMockSetMultipleMongoData(testMap)(Left(new Exception()))
             testInternalServerErrors(isAgent = false, isChange = false, id = Some(testSelfEmploymentId))
           }
         }
         s"SelfEmployment - unable to set session data ${CeaseIncomeSourceData.incomeSourceIdField}" when {
           "called .submit" in {
             setupMockCreateSession(true)
-            setupMockSetSessionKeyMongo(CeaseIncomeSourceData.dateCeasedField)(Right(true))
-            setupMockSetSessionKeyMongo(CeaseIncomeSourceData.incomeSourceIdField)(Left(new Exception()))
+            val testMap: Map[String, String] = Map(
+              CeaseIncomeSourceData.dateCeasedField -> "2022-08-27",
+              CeaseIncomeSourceData.incomeSourceIdField -> testSelfEmploymentId
+            )
+            setupMockSetMultipleMongoData(testMap)(Right(true))
+            setupMockSetMultipleMongoData(testMap)(Left(new Exception()))
```

Comment:
this second mock will overwrite the first mock. in the code I think an exception would be returned if setMultipleMongoData can't set either field, so you can just remove

`            setupMockSetMultipleMongoData(testMap)(Right(true))
`